scripts/withAppEnvs/index.js

```diff
@@ -0,0 +1,50 @@
+const path = require('path');
+const fs = require('fs');
+const dotenv = require('dotenv');
+const buildEnvTypeFile = require('./buildEnvTypeFile');
+
+function getAppEnvs() {
+    const projectDir = process.cwd();
+  
+    let APP_ENV = process.env.APP_ENV;
+  
+    if (!APP_ENV) {
+      console.warn('\x1b[33m%s\x1b[0m', '⚙️ APP_ENV is not set. Using production');
+  
+      // We use production as default because it will be used
+      // by e.g. Vercel and Netlify when building the app via `npm run build`
+      APP_ENV = 'production';
+    }
+  
+    const envFilePath = path.join(projectDir, 'env', `.env.${APP_ENV}`);
+  
+    if (!fs.existsSync(envFilePath)) {
+      throw new Error(`.env.${APP_ENV} file not found`);
+    }
+  
+    // Create object from .env file
+    const envsObject = dotenv.parse(fs.readFileSync(envFilePath));
+  
+    console.log('\x1b[33m%s\x1b[0m', '⚙️ Using env file:', envFilePath);
```

Comment:
ogolnie tutaj mialem taki problem w projekcie ze w next.config nie mialem dostepnych zaladowanych envow w process.env (potrzebowalem jakis publicznych urli do konfiguracji rewriteow). Oczywiscie moglem korzsytac ze zmiennej envs ale jak bym chcial uzyc do generowania rewritow jakies funkcji to musial bym przekazywac do niej envs lub jakies wartosci z tych envow.

moze by dodac:
```
const envs = {
      APP_ENV: APP_ENV,
      ...envsObject,
    };

process.env = {
...process.env,
...envs
}

return envs;
```